src/bom/extract.rs

```diff
@@ -8,14 +8,36 @@ use crate::bom::bitcode::ELF_SECTION_NAME;
 #[derive(thiserror::Error,Debug)]
 pub enum ExtractError {
     #[error("Error running command {0:} {1:?} ({2:?})")]
-    ErrorRunningCommand(String,Vec<OsString>,std::io::Error)
+    ErrorRunningCommand(String,Vec<OsString>,std::io::Error),
+
+    #[error("Temp directory lost during extraction")]
+    ErrorLostTmpDir,
 }
 
 pub fn extract_bitcode_entrypoint(extract_options : &ExtractOptions) -> anyhow::Result<i32> {
     let tmp_dir = tempfile::TempDir::new()?;
+    let res = do_bitcode_extraction(extract_options, tmp_dir.path());
+
+    // The tmp_dir should always exist, so this should never execute the body of
+    // the if statement. However, the point of this is that *the tmp_dir should
+    // still exist*, and part of this is avoiding the "Early drop pitfall"
+    // described at https://docs.rs/tempfile/latest/tempfile, so this if
+    // expression uses the original tmp_dir thus allowing Rust to help us ensure
+    // it isn't dropped somewhere in do_bitcode_extraction by an inadvertent
+    // Copy/move that would trigger resource cleanup in disposal of this original
+    // object.
+    if !tmp_dir.path().exists() {
+        return Err(anyhow::Error::new(ExtractError::ErrorLostTmpDir));
+    };
```

Comment:
This is a good problem to be concerned about, but I wonder if we can do it in a more robust way.

If we just call `std::mem::drop(tmp_dir);` here (see https://doc.rust-lang.org/std/mem/fn.drop.html), that will forcibly take ownership here. That implies we would get a compile error if we moved it before this point. It happens to be the case that right now we don't, but this would force it.